packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1336,7 +1336,7 @@ class _TabBarViewState extends State<TabBarView> {
     _warpUnderwayCount += 1;
     if (notification is ScrollUpdateNotification && !_controller!.indexIsChanging) {
       if ((_pageController.page! - _controller!.index).abs() > 1.0) {
-        _controller!.index = _pageController.page!.floor();
+        _controller!.index = _pageController.page!.round();
         _currentIndex =_controller!.index;
       }
       _controller!.offset = (_pageController.page! - _controller!.index).clamp(-1.0, 1.0);
```

Comment:
It seems to change the behavior of the offset, does it?

Reply:
Yes it does. To be precise, I adapted the behavior to the end of flinging. 
 
When flinging ends, `round` is used.
https://github.com/flutter/flutter/blob/7620e5d33e2589391c2888cea750f08415573ccf/packages/flutter/lib/src/material/tabs.dart#L1344